<p>Relative imports caused a <code>None</code>-entry to be created for <code>nr.c4d</code> in <code>sys.modules</code> before <code>nr.__path__</code> was extended a second time. This is the issue I opened on my <code>localimport</code> repository:</p>

<blockquote>
  <p><strong>Scenario</strong>: <code>nr.procedural</code> does <code>import c4d</code>. If <code>nr.c4d</code> does not exist, the module will be added as a None-entry to <code>sys.modules</code>. If then <code>_localimport</code> is used to import <code>nr.c4d</code>, the module will not be found as the module cache says that the module does not exist (by the None-entry).</p>
  
  <p><strong>Solution</strong>: <code>_localimport</code> must delete None-entries for namespace packages on <code>__enter__()</code>.</p>
</blockquote>

<p>The <a href="https://github.com/NiklasRosenstein/localimport/commit/842d2aee2b37782441c53143d46864a597e080d4" rel="nofollow">fix</a> was relatively easy.</p>

<p>I will  need to add <code>from __future__ import absolute_import</code> to all <code>nr.</code> sub-packages that <code>import c4d</code> though.</p>
